### PR TITLE
PHP 8.3 support added

### PR DESCRIPTION
**Pull request by glo71317**
|    Q          |   A
|-------------- | ------
| Documentation | yes/no
| Bugfix        | yes/no
| BC Break      | yes/no
| New Feature   | yes/no
| RFC           | yes/no
| QA            | yes/no

### CODE REVIEW

**Comment by glo71317** [comment]
@froschdesign Can you please review the PR? We are looking for your help to merge this because we are working on Magento Dependencies and support php 8.3 due to laminas dependencies we are blocked so looking forward from your side. Can you please do the needful?

File: src/Figlet/Figlet.php [hunk 1]
```diff
@@ -832,6 +832,10 @@ protected function _smushAmount()
             if (empty($leftChar) || $leftChar === ' ') {
                 $amount++;
             } elseif (! empty($rightChar)) {
+                /**
+                 * @psalm-suppress RedundantConditionGivenDocblockType
```
Comment (Ocramius) [inline src/Figlet/Figlet.php]: We should baseline these, not suppress them

File: src/Table/DecoratorManager.php [hunk 2]
```diff
@@ -46,7 +46,10 @@ class DecoratorManager extends AbstractPluginManager
         'zendtexttabledecoratorunicode' => Decorator\Unicode::class,
     ];
 
-    /** @var FactoriesConfigurationType */
+    /**
+     * @var FactoriesConfigurationType
+     * @psalm-suppress NonInvariantDocblockPropertyType
```
Comment (Ocramius) [inline src/Table/DecoratorManager.php]: This is fixable: doesn't need suppression

File: src/Table/Table.php [hunk 3]
```diff
@@ -499,6 +499,7 @@ public function render()
     /**
      * Magic method which returns the rendered table
      *
+     * @psalm-suppress MethodSignatureMustProvideReturnType
```
Comment (Ocramius) [inline src/Table/Table.php]: This should be baselined, not suppressed

File: test/MultiByteTest.php [hunk 4]
```diff
@@ -9,6 +9,7 @@
 
 /**
  * @group      Laminas_Text
+ * @psalm-suppress DeprecatedMethod
```
Comment (Ocramius) [inline test/MultiByteTest.php]: To be baselined, not suppressed

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by froschdesign** [comment]
@gsteel 
Can you please take a look at this? Are the extensions to Psalm correct?

File: src/Figlet/Figlet.php (hunk 1 above)
Comment (glo17680) [reply src/Figlet/Figlet.php]: Made all the suggested changes and pushed the code. Kindly review. Thank you.

File: src/Table/DecoratorManager.php (hunk 2 above)
Comment (glo17680) [reply src/Table/DecoratorManager.php]: Done

File: src/Table/Table.php (hunk 3 above)
Comment (glo17680) [reply src/Table/Table.php]: Done

File: test/MultiByteTest.php (hunk 4 above)
Comment (glo17680) [reply test/MultiByteTest.php]: Done

**Comment by glo71317** [comment]
@Ocramius @gsteel PR is ready more than a week, No one PR is reviewing Can you please help us to review and merge this?

**Review by Ocramius — APPROVED** [review]
LGTM

About timeline: nobody (that I know of) is paid to work on this over here, and we do it in our free time/when we can/out of our good will, so please be patient, as our only timeline is "when we want to do it between work and life" :P

**Pull request merged**